oshdb/src/test/java/org/heigit/ohsome/oshdb/contribution/ContributionsRelationTest.java

```diff
@@ -0,0 +1,315 @@
+package org.heigit.ohsome.oshdb.contribution;
+
+import static java.util.Collections.emptyList;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.CREATION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.DELETION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.GEOMETRY_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.MEMBER_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.ROLE_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.TAG_CHANGE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.EnumSet;
+import java.util.List;
+import org.heigit.ohsome.oshdb.OSHDBTest;
+import org.junit.jupiter.api.Test;
+
+class ContributionsRelationTest extends OSHDBTest {
+
+  @Test
+  void testMemberTagChanges() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(3, 3500, 253, 3, tags(), 1, 1),
+                node(2, 2500, 253, 3, tags(), 1, 1), // squash
+                node(1, 1100, 101, 1, tags(), 0, 0))), // minor-major change
+            way(1, 1000, 101, 1, tags(), mems(1))));
+
+    var versions = relations(1,
+        relation(2, 2000, 202, 2, tags(tag(1, 1)), mems(w(1, 0))),
+        relation(1, 1000, 101, 1, tags(), mems(w(1, 0), n(1, 0))));
+    var osh = osh(versions, nodes, ways);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(3500, 253, 3, versions.get(0), contrib);
+    assertEquals(EnumSet.of(GEOMETRY_CHANGE), contrib.getMinorTypes());
+
+    assertTrue(contribs.hasNext());
+    contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(TAG_CHANGE, MEMBER_CHANGE), contrib.getTypes());
+
+    assertTrue(contribs.hasNext());
+    contrib = contribs.next();
+    assertContrib(1100, 101, 1, versions.get(1), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testOutsideMax() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(1, 1000, 101, 1, tags(), 0, 0))),
```

Comment:
Yes, but that the way use the some node as the relation as well, was not the intention here, but if it would, that this would make totally sense. 
We can change it to the suggested version, or use a different node id for the way.
What do you prefer?